app/src/app/pages/plando-page/plando-save-load/plando-save-load.component.ts

```diff
@@ -62,11 +68,118 @@ export class PlandoSaveLoadComponent implements OnInit {
+          }
+          if (typeof v !== 'object' || (Array.isArray(v) && v.length > 0)) {
+            return [[k, v]];
+          }
+          v = tidyObj(v);
+          if (Object.keys(v).length === 0) {
+            return []
+          } else {
+            return [[k, v]];
+          }
+        }));
+      }
+      const a = document.createElement('a');
+      const plandoFile = new Blob([JSON.stringify(tidyObj(plandoFormObj))], { type: 'application/json' });
+      a.href = URL.createObjectURL(plandoFile);
+      a.download = name + '.json';
+      a.click();
+    }
+  }
+
+  public deletePlandoSettings(name: string, notFound: boolean = false) {
     this.savedPlandoNames.delete(name);
-    localStorage.removeItem(name);
-    localStorage.setItem(this.SAVED_PLANDO_NAMES_KEY, Array.from(this.savedPlandoNames).join(','));
+    localStorage.removeItem(SAVED_PLANDO_NAME_PREFIX + name);
+    localStorage.setItem(SAVED_PLANDO_NAMES_KEY, Array.from(this.savedPlandoNames).join(','));
     this.lastPlandoName = name;
-    this.saveLoadStatus = 'deleted';
+    this.saveLoadStatus = notFound ? 'notFound' : 'deleted';
+  }
+
+  public importPlandoSettings() {
+    const input = document.createElement('input');
+    input.type = 'file';
+    input.accept = 'application/json';
+    input.onchange = async () => {
+      if (!input.files || !input.files.item(0))
+        return;
+      try {
+        const file = input.files.item(0) as File;
+        if (!file.name.endsWith('.json')) {
+          throw new Error('invalid');
+        }
+        const plandoName = file.name.substring(0, file.name.length - 5).replace(/,/g, '').substring(0, 20);
+        if (this.savedPlandoNames.has(plandoName)) {
+          throw new Error('exists');
+        }
+        const fileContents = await this.loadFile(file);
+        // Because we strip exported JSON, we can't just set value on the formgroup.
+        // Merge the object into the default formgroup object we save on page load.
+        // As a small bonus, this works as a sort of ad-hoc validator.
+        const importedObj = JSON.parse(fileContents);
+        const defaultObj = JSON.parse(localStorage.getItem(DEFAULT_PLANDOMIZER_KEY));
+        const mergeObjectDeep = (target, source) => {
+          for (const key in source) {
+            if (target[key] === undefined) {
+              throw new Error('invalid');
+            }
+            if (typeof source[key] === 'object' && !Array.isArray(source[key])) {
+              mergeObjectDeep(target[key], source[key]);
+            } else {
+              Object.assign(target, { [key]: source[key] });
+            }
+          }
+        }
+        mergeObjectDeep(defaultObj, importedObj);
+        this.plandoFormGroup.setValue(defaultObj);
+        this.importStatus = 'success';
+        this.saveLoadFormGroup.get('plandoName').setValue(plandoName);
+        this.savePlandoSettings(plandoName);
+      } catch (e) {
+        if (e.message && e.message.length <= 7) {
```

Comment:
Why the length check here?

Reply:
Oops. Relic of an earlier version where I was also catching errors thrown by `FormGroup.setValue`. That's not happening anymore, so I'll remove it.